python/cudf/cudf/utils/dtypes.py

```diff
@@ -235,3 +235,39 @@ def min_numeric_column_type(x):
         return np.promote_types(max_bound_dtype, min_bound_dtype)
 
     return x.dtype
+
+
+def check_upcast_unsupported_dtype(dtype):
```

Comment:
I would rename to `check_cast_unsupported_dtype`. We downcast in certain situations, i.e. `uint64` --> `int64`.

Reply:
Ah, Right 👍

Reply:
Renamed it